pyftdi/i2c.py

```diff
@@ -1041,15 +1041,18 @@ def _do_epilog(self) -> None:
         # be sure to purge the MPSSE reply
         self._ftdi.read_data_bytes(1, 1)
 
-    def _send_check_ack(self, cmd: bytearray):
+    def _send_check_ack(self, cmd: bytearray, fake_tristate_ack_end_as_output: bool):
         # note: cmd is modified
         if self._fake_tristate:
             # SCL low, SDA high-Z (input)
             cmd.extend(self._clk_lo_data_input)
             # read SDA (ack from slave)
             cmd.extend(self._read_bit)
-            # leave SCL low, restore SDA as output
-            cmd.extend(self._clk_lo_data_hi)
+            # after ACK, if writing next, end as output. if reading next, end
+            #  as an input to prevent a short as the slave may pull SDA low after ACK
+            if fake_tristate_ack_end_as_output:
+                # leave SCL low, restore SDA as output
+                cmd.extend(self._clk_lo_data_hi)
```

Comment:
Probably could be done way easier:

```py
if self._fake_tristate:
  cmd.extend(self._clk_lo_data_input)
else:
  cmd.extend(self._clk_lo_data_hi)
```

Reply:
erm, wait... oops... _fake_tristate is already checked, so the only thing needed is dropping line 1052 to don't restore the output

Reply:
You're looking at an old commit. I did simplify the implementation that does away with the additional parameters. Please make sure you choose the option to show all changes or click the "Files changed" tab above.